### DISCUSSION TITLE
TypeScript + Recipes

### DISCUSSION THREAD

**Discussion by altano** [chakra-ui/panda#991] [Q&A] [answered]
In the [TypeScript Guide](https://panda-css.com/docs/concepts/recipes#typescript-guide) you show how to use the helper `RecipeVariantProps` type to extract the variant types, e.g.:

For some reason this helper has a much looser type locally for me than the example docs and the playground:

Playground ([link](https://play.panda-css.com/2wYmL-Z38K)):
<img width="330" alt="image" src="https://github.com/chakra-ui/panda/assets/1009/33498a6a-a336-4aba-bf94-7e61b54c4a0f">

Local:
<img width="437" alt="image" src="https://github.com/chakra-ui/panda/assets/1009/4dcf10d8-3036-4691-95da-5f44e6834ba0">

I'm using panda v0.6.0 and typescript v5.1.3 and I couldn't find any interesting differences in panda or typescript's configs so I'm not sure what is causing the difference. I think it boils down to the fact that `Parameters` gives me undefined here:

<img width="465" alt="image" src="https://github.com/chakra-ui/panda/assets/1009/b659b65d-07c2-4ea1-a373-d2e5c9d896aa">

Any idea what might be different between the documentation and the playground and my local environment?

The workaround is simple: I have to wrap my type in `NonNullable`, and maybe `Required` if I want it to be required. This is what my final prop type looks like:

```tsx
type ButtonVariants = Required<NonNullable<RecipeVariantProps<typeof buttonStyle>>>;
...
size: ButtonVariants["size"];
```

**Comment by altano** [comment]
FWIW, the behavior I'm seeing locally is actually what I'd expect, because of [these types](https://github.com/chakra-ui/panda/blob/ac719b8b4c351be773ec5a209fa00793e367e255/packages/types/src/recipe.ts#L9-L15) which clearly make the props argument and all its keys optional. The real mystery is why the playground works as desired?

**Comment by anubra266** [comment, marked as answer]
@altano check this type if they're the same
![CleanShot 2023-07-10 at 01 38 41@2x](https://github.com/chakra-ui/panda/assets/30869823/1fce5d5c-5d09-4de0-8834-e1a56fcbe36a)

The playground uses dev version of panda, so it's possible the types are changed but unreleased

**Reply by altano** [reply]
Yep, that's the difference!

Playground:
<img width="606" alt="image" src="https://github.com/chakra-ui/panda/assets/1009/ea4eb951-9cd8-429f-a5db-7d71f4081655">

v0.6.0:
https://github.com/chakra-ui/panda/blob/%40pandacss/types%400.6.0/packages/types/src/recipe.ts#L17

**Reply by altano** [reply]
If the version in the playground is on its way to release eventually, we can go ahead and close this discussion?

**Comment by abhayzmp** [comment]
This issue also doesn't allow the storybook stories to render its controls
